source/Stargate-Metrics-Tests/MetricsRESTfulControllerTest.class.st

```diff
@@ -34,27 +34,26 @@ MetricsRESTfulControllerTest >> testGetMetricsInJson [
 	| response |
 
 	response := resourceController
-		getMetricsBasedOn:
-			( self
-				requestToGET: self resourceUrl
-				accepting: resourceController operationalMetricsVersion1dot0dot0MediaType )
-		within: self newHttpRequestContext.
+		            getMetricsBasedOn: ( self
+				              requestToGET: self resourceUrl
+				              accepting: resourceController operationalMetricsVersion1dot0dot0MediaType )
+		            within: self newHttpRequestContext.
 
 	self
 		assert: response isSuccess;
 		assert: response contentType asMediaType
-			equals: resourceController operationalMetricsVersion1dot0dot0MediaType;
-		withJsonFromContentsIn: response
-			do: [ :metrics | 
-			self assert: metrics size equals: 7.
-			metrics
-				do: [ :metric | 
-					self
-						assert: metric name notEmpty;
-						assert: metric description notEmpty;
-						assert: metric value isNumber;
-						assert: ( #('Gauge' 'Counter') includes: metric type )
-					]
+		equals: resourceController operationalMetricsVersion1dot0dot0MediaType;
+		withJsonFromContentsIn: response do: [ :metrics |
+			self
```

Comment:
Maybe here and I saw the same as part of: MetricsPluginTest >> testConfiguredRunningAllMetrics include an example of what the metrics look like in each case so that it's not just assert size equals X.

Reply:
The problem here is that the metric values will change on every test run, the number and kind of metrics are different in every platform and the Prometheus format is not easy to test without controlling all the variables.
testGetMetricsInJson does a bit of what you're asking here by testing the number of metrics available in every platform, and the general structure of each one but without testing the exact values.